# Target repository: external/susfs4ksu
# Target: crDroid 17.0
# Rebase patch context only; kernel additions and removals stay identical to upstream.
diff --git a/kernel_patches/50_add_susfs_in_gki-android15-6.6.patch b/kernel_patches/50_add_susfs_in_gki-android15-6.6.patch
--- a/kernel_patches/50_add_susfs_in_gki-android15-6.6.patch
+++ b/kernel_patches/50_add_susfs_in_gki-android15-6.6.patch
@@ -1,6 +1,5 @@
-
 diff --git a/drivers/input/input.c b/drivers/input/input.c
-index 9bb1d3de7..a6333655c 100644
+index 9bb1d3de723ee133a24a4a82d06b18b93eb84e63..a6333655c9067893408ed0dd546758dd4e4300ce 100644
 --- a/drivers/input/input.c
 +++ b/drivers/input/input.c
 @@ -391,6 +391,12 @@ static void input_event_dispose(struct input_dev *dev, int disposition,
@@ -28,7 +27,7 @@
  		if (type != EV_SYN)
  			add_input_randomness(type, code, value);
 diff --git a/fs/Makefile b/fs/Makefile
-index c6963a9a4..e27ac368b 100644
+index c6963a9a4d0ae021b5bc33e63413c2e926d1a00b..e27ac368bbb4b07825b831ae408f4aa95e368725 100644
 --- a/fs/Makefile
 +++ b/fs/Makefile
 @@ -22,6 +22,8 @@ obj-y :=	open.o read_write.o file_table.o super.o \
@@ -41,7 +40,7 @@
  obj-$(CONFIG_PROC_FS)		+= proc_namespace.o
  obj-$(CONFIG_LEGACY_DIRECT_IO)	+= direct-io.o
 diff --git a/fs/exec.c b/fs/exec.c
-index 8dc93215b..339c85ea6 100644
+index 8dc93215b9c9b86c503eff653ed3e821a0b2ba4f..339c85ea602c5b8c0cbe8946defa74132890dc7e 100644
 --- a/fs/exec.c
 +++ b/fs/exec.c
 @@ -67,6 +67,9 @@
@@ -145,7 +144,7 @@
  	free_bprm(bprm);
  
 diff --git a/fs/namei.c b/fs/namei.c
-index b31b583f1..95dd56a79 100644
+index b31b583f18789cc8043770f0ed99b03e1c4856a8..95dd56a7936327f5eb79e837c12f8746d6e7ade4 100644
 --- a/fs/namei.c
 +++ b/fs/namei.c
 @@ -41,10 +41,21 @@
@@ -585,7 +584,7 @@
  	do_delayed_call(&done);
  	return res;
 diff --git a/fs/namespace.c b/fs/namespace.c
-index 97fc42277..51b85060f 100644
+index 97fc4227773ff4f5255306892934daf329815604..51b85060f6df1979cdb8d3bcb0618a3b3fe75eef 100644
 --- a/fs/namespace.c
 +++ b/fs/namespace.c
 @@ -32,12 +32,22 @@
@@ -921,7 +920,7 @@
 +}
 +#endif // #ifdef CONFIG_KSU_SUSFS
 diff --git a/fs/notify/fdinfo.c b/fs/notify/fdinfo.c
-index 1aa7de550..19a20061e 100644
+index 1aa7de55094cd98a123ec3b23147759a7f013784..19a20061e60d77bcf7dd7e253926899335a31bbe 100644
 --- a/fs/notify/fdinfo.c
 +++ b/fs/notify/fdinfo.c
 @@ -12,6 +12,9 @@
@@ -1042,7 +1041,7 @@
  			   inode_mark->wd, inode->i_ino, inode->i_sb->s_dev,
  			   inotify_mark_user_mask(mark));
 diff --git a/fs/open.c b/fs/open.c
-index 0641cb3fd..2a979fdff 100644
+index b799e77848ee2abdf6c0a2e40af105c08ac4f11c..ce9f673056b1aee82f258064de901d23cab16681 100644
 --- a/fs/open.c
 +++ b/fs/open.c
 @@ -34,6 +34,9 @@
@@ -1103,7 +1102,7 @@
  		goto out;
  
 diff --git a/fs/proc/base.c b/fs/proc/base.c
-index cf20a8f08..71ee077cf 100644
+index cf20a8f08d606618448cc42ef3aed6b052faf627..71ee077cff467fc4ee71acb68a52a7abe1d11393 100644
 --- a/fs/proc/base.c
 +++ b/fs/proc/base.c
 @@ -101,6 +101,10 @@
@@ -1159,7 +1158,7 @@
  			continue;
  
 diff --git a/fs/proc/bootconfig.c b/fs/proc/bootconfig.c
-index 2e244ada1..ca76c06d0 100644
+index 2e244ada1f970b40dfc11417e5fd43579074d42f..ca76c06d0bbffbe8ed185d79c8e52d77f5d520eb 100644
 --- a/fs/proc/bootconfig.c
 +++ b/fs/proc/bootconfig.c
 @@ -12,8 +12,21 @@
@@ -1185,7 +1184,7 @@
  		seq_puts(m, saved_boot_config);
  	return 0;
 diff --git a/fs/proc/fd.c b/fs/proc/fd.c
-index 4297287f6..0ce657709 100644
+index 4297287f6ca09d6ecbace9f521ad741ee63505a8..0ce657709b30b1c5abcfbc05c3944d191d8958f2 100644
 --- a/fs/proc/fd.c
 +++ b/fs/proc/fd.c
 @@ -15,11 +15,22 @@
@@ -1284,7 +1283,7 @@
  	show_fd_locks(m, file, files);
  	if (seq_has_overflowed(m))
 diff --git a/fs/proc/task_mmu.c b/fs/proc/task_mmu.c
-index 052e6d7bc..24a441252 100644
+index 052e6d7bc62d21d90f29935b31c7624762411c1a..24a4412525a32836012d13ca7a69dc7301bec4d0 100644
 --- a/fs/proc/task_mmu.c
 +++ b/fs/proc/task_mmu.c
 @@ -23,6 +23,9 @@
@@ -1422,7 +1421,7 @@
  		start_vaddr = end;
  
 diff --git a/fs/proc_namespace.c b/fs/proc_namespace.c
-index 250eb5bf7..444c8f88f 100644
+index 250eb5bf7b522b16476165de34917c5dbc2f8365..444c8f88f1d0254ffba54bc57fcd648d2b8021bc 100644
 --- a/fs/proc_namespace.c
 +++ b/fs/proc_namespace.c
 @@ -12,12 +12,20 @@
@@ -1641,7 +1640,7 @@
  }
  
 diff --git a/fs/read_write.c b/fs/read_write.c
-index 066490235..2f4ff91be 100644
+index 06649023585c17b0031ae066d0d1b8ca8f819e49..2f4ff91be4daa789e05c6929a053e0a5d36d45bf 100644
 --- a/fs/read_write.c
 +++ b/fs/read_write.c
 @@ -619,8 +619,18 @@ ssize_t ksys_read(unsigned int fd, char __user *buf, size_t count)
@@ -1664,7 +1663,7 @@
  }
  
 diff --git a/fs/readdir.c b/fs/readdir.c
-index c8c46e294..a18bb8379 100644
+index c8c46e29443139d1e496bbb2236e4faf6af9effa..a18bb8379a21ecd174c2f4c7a38898181a24604d 100644
 --- a/fs/readdir.c
 +++ b/fs/readdir.c
 @@ -24,6 +24,10 @@
@@ -1941,7 +1940,7 @@
  	if (error >= 0)
  		error = buf.error;
 diff --git a/fs/stat.c b/fs/stat.c
-index 517fa4ed8..1df35c064 100644
+index 517fa4ed885e82926be81e5949a86d71013e2b05..1df35c06437a7d855d72570e4c40834064abfe3e 100644
 --- a/fs/stat.c
 +++ b/fs/stat.c
 @@ -20,12 +20,27 @@
@@ -2074,7 +2073,7 @@
  
  	if (path.mnt->mnt_root == path.dentry)
 diff --git a/fs/statfs.c b/fs/statfs.c
-index 2a37f9f3d..05d7dfab1 100644
+index 2a37f9f3dba1f421cced281056906298024e71be..05d7dfab1ad75f15495c4c5bc489c343d69adbba 100644
 --- a/fs/statfs.c
 +++ b/fs/statfs.c
 @@ -9,8 +9,20 @@
@@ -2182,7 +2181,7 @@
  	if (!error)
  		buf->f_flags = calculate_f_flags(path->mnt);
 diff --git a/fs/super.c b/fs/super.c
-index 2d8303d32..505e62bb8 100644
+index 2d8303d32ccf6097609b7b1310363ac50a044fd4..505e62bb88b6338b02239bd5286b1305083d2607 100644
 --- a/fs/super.c
 +++ b/fs/super.c
 @@ -37,6 +37,9 @@
@@ -2231,7 +2230,7 @@
  	 * Many userspace utilities consider an FSID of 0 invalid.
  	 * Always return at least 1 from get_anon_bdev.
 diff --git a/kernel/kallsyms.c b/kernel/kallsyms.c
-index cd44e9c42..4c6c4c74d 100644
+index cd44e9c427533015dc1e3ad32817d3242c040f4b..4c6c4c74dbb42d0b5cb47be6f7b887d394071fc0 100644
 --- a/kernel/kallsyms.c
 +++ b/kernel/kallsyms.c
 @@ -31,6 +31,9 @@
@@ -2282,7 +2281,7 @@
  }
  
 diff --git a/kernel/reboot.c b/kernel/reboot.c
-index 344ceaf5e..c2d1a782d 100644
+index 344ceaf5ed130975305528f90b8ab0a802202b16..c2d1a782d5a54fbd311e1e0a7dab2408c40909c4 100644
 --- a/kernel/reboot.c
 +++ b/kernel/reboot.c
 @@ -693,6 +693,9 @@ EXPORT_SYMBOL_GPL(kernel_power_off);
@@ -2312,7 +2311,7 @@
  	if (!ns_capable(pid_ns->user_ns, CAP_SYS_BOOT))
  		return -EPERM;
 diff --git a/kernel/sys.c b/kernel/sys.c
-index c6b0141e4..5a9bf1c58 100644
+index c6b0141e413544a3f0522e9e03d1907fdad6b841..5a9bf1c5846e4a98160d117a5dcc8c8d14d973db 100644
 --- a/kernel/sys.c
 +++ b/kernel/sys.c
 @@ -661,6 +661,9 @@ SYSCALL_DEFINE1(setuid, uid_t, uid)
@@ -2358,7 +2357,7 @@
  	if (copy_to_user(name, &tmp, sizeof(tmp)))
  		return -EFAULT;
 diff --git a/mm/memory.c b/mm/memory.c
-index 1f8011193..b5ea74223 100644
+index 1f8011193ac2793f280dac369bc961ad6178cf1b..b5ea7422342b14f9f97289355364a61ae7c090b8 100644
 --- a/mm/memory.c
 +++ b/mm/memory.c
 @@ -82,6 +82,9 @@
@@ -2391,7 +2390,7 @@
  		if (IS_ERR_OR_NULL(page)) {
  			/* We might need to expand the stack to access it */
 diff --git a/security/selinux/avc.c b/security/selinux/avc.c
-index 8e40b0941..629658620 100644
+index 8e40b0941d1c8f05fbc338f78b9483d05e862894..629658620f8960424c1e78dbe4650a4545638e51 100644
 --- a/security/selinux/avc.c
 +++ b/security/selinux/avc.c
 @@ -48,6 +48,12 @@
@@ -2434,7 +2433,7 @@
  	audit_log_format(ab, " tclass=%s", tclass);
  
 diff --git a/security/selinux/hooks.c b/security/selinux/hooks.c
-index 68d1f244e..2691f5abc 100644
+index 3f4c1bfec0160268a4048710b02ad8fe3c29e880..96374d4b18dbac92a18cb010e1f4d521a9234762 100644
 --- a/security/selinux/hooks.c
 +++ b/security/selinux/hooks.c
 @@ -107,6 +107,10 @@
@@ -2448,7 +2447,7 @@
  
  /* SECMARK reference count */
  static atomic_t selinux_secmark_refcount = ATOMIC_INIT(0);
-@@ -6513,6 +6517,17 @@ static int selinux_setprocattr(const char *name, void *value, size_t size)
+@@ -6624,6 +6628,17 @@ static int selinux_setprocattr(const char *name, void *value, size_t size)
  	return error;
  }
  
@@ -2466,7 +2465,7 @@
  static int selinux_ismaclabel(const char *name)
  {
  	return (strcmp(name, XATTR_SELINUX_SUFFIX) == 0);
-@@ -7138,8 +7153,11 @@ static struct security_hook_list selinux_hooks[] __ro_after_init = {
+@@ -7252,8 +7267,11 @@ static struct security_hook_list selinux_hooks[] __ro_after_init = {
  	LSM_HOOK_INIT(d_instantiate, selinux_d_instantiate),
  
  	LSM_HOOK_INIT(getprocattr, selinux_getprocattr),
@@ -2480,7 +2479,7 @@
  	LSM_HOOK_INIT(secctx_to_secid, selinux_secctx_to_secid),
  	LSM_HOOK_INIT(release_secctx, selinux_release_secctx),
 diff --git a/security/selinux/selinuxfs.c b/security/selinux/selinuxfs.c
-index b8c263457..5842481c8 100644
+index b8c2634579116ad6ca5d0c05ef85fe064097ec53..5842481c8956f40990e9c6c87af7ad5630980d88 100644
 --- a/security/selinux/selinuxfs.c
 +++ b/security/selinux/selinuxfs.c
 @@ -44,6 +44,16 @@
